tests/03_component.spec.tsx

```diff
@@ -15,27 +14,27 @@ type ExtractState<S> = S extends {
   : never;
 
 function createZustandContext<Store extends StoreApi<unknown>>(
-  makeStore: () => Store,
+  initializeStore: () => Store,
 ) {
-  const Context = createContext<Store | undefined>(undefined);
+  const Context = createContext<Store | null>(null);
   const StoreProvider = ({ children }: { children: ReactNode }) => {
-    const [store] = useState(makeStore);
+    const store = useRef(initializeStore()).current;
```

Comment:
this will call `initializeStore` every time `StoreProvider` renders, which is unnecessary work

it should either be a [useState with a lazy initializer](https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state) or a [conditional ref initialization](https://react.dev/reference/react/useRef#avoiding-recreating-the-ref-contents), both of which avoid this

Reply:
I knew it, and I don't do it in the library code, but yeah, as people read the test code, let me fix it.

Reply:
#7